Add plugin: Nutstore Sync

miaobuao opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin:

## Release Checklist
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [x]  Linux
  - [x]  Android _(if applicable)_
  - [x]  iOS _(if applicable)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[# 🔄 Obsidian Nutstore](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/README.md#L1)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

[private statusHideTimer: NodeJS.Timeout | null = null](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/index.ts#L35)
Use a regular number as the type here and use `window.setTimeout` / `window.clearTimeout` where this is being used.

[const styleEl = document.getElementById('nutstore-sync-styles')](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/index.ts#L172-L178)
Remove this code, it's not needed.

[containerEl.createEl('h2', { text: i18n.t('settings.title') })](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/settings.ts#L47)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[import { basename, isAbsolute, join } from 'path'](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/utils/traverse-local-vault.ts#L4), [import path from 'path'](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/utils/std-remote-path.ts#L1), [import { isAbsolute } from 'path'](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/utils/stat-webdav-item.ts#L1), [import { basename } from 'path'](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/utils/stat-vault-item.ts#L2), [import { isAbsolute, relative } from 'path'](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/utils/remote-path-to-local-path.ts#L2), [import { dirname, normalize } from 'path'](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/utils/mkdirs-vault.ts#L2), [import { normalize } from 'path'](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/utils/is-sub.ts#L1), [import { normalize } from 'path'](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/utils/get-root-folder-name.ts#L1), [import path from 'path'](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/sync/index.ts#L4), [import path from 'path'](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/sync/tasks/task.interface.ts#L2), [import { basename, join } from 'path'](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/api/webdav.ts#L3)
These imports are only available from a Node.js runtime, which means that this will throw errors on Mobile (that's a problem since you have `isDesktopOnly` marked as `false` in your manifest.json file.

[import dayjs from 'dayjs'](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/sync/index.ts#L2), [import dayjs from 'dayjs'](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/sync/tasks/conflict-resolve.task.ts#L2)
Instead of importing `dayjs` you should use the `moment` instance that is bundled with Obsidian.
It can be accesed like this: `import { moment } from "obsidian";`

[name: 'Check Connection',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L15), [name: 'Remote Directory',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L23), [loggedIn: 'Logged In',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L40-L41), [name: 'Use Git-style Conflict Markers',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L46), [name: 'Backup Warning',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L50), [name: 'Conflict Resolution Strategy',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L54), [diffMatchPatch: 'Smart Merge (Recommended)',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L56-L57), [ctions: {](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L59-L61), [confirmTitle: 'Confirm Logout',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L64), [confirm: 'Confirm Logout',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L67), [startButton: 'Start Sync',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L91-L93), [cancelButton: 'Cancel Sync',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L96), [title: 'Sync Confirmation',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L99), [confirm: 'Confirm Sync',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L102), [remoteDir: 'Remote Directory: {{dir}}',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L104-L105), [newFolder: 'New Folder',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/packages/webdav-explorer/src/i18n/locales/en.ts#L2), [goBack: 'Go Back',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/packages/webdav-explorer/src/i18n/locales/en.ts#L3), [currentPath: 'Current Path',](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/packages/webdav-explorer/src/i18n/locales/en.ts#L6)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[ctions: {](https://github.com/miaobuao/obsidian-nutstore-sync/blob/9ef2ece678e7af9c48bf60802c24bd2eb91f760b/src/i18n/locales/en.ts#L59-L61)
[Avoid "settings" in settings headings](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+%22settings%22+in+settings+headings)

miaobuao commented:
@joethei Thank you for your detailed review! I have addressed all the issues you raised and made the necessary changes to the plugin. Regarding the "browser" field in package.json, I have configured it to ensure mobile compatibility by mapping Node.js-specific modules to their browser-compatible versions.

https://github.com/miaobuao/obsidian-nutstore-sync/blob/327908e2185914a7904349c4ac87613440207726/package.json#L56

https://github.com/miaobuao/obsidian-nutstore-sync/blob/327908e2185914a7904349c4ac87613440207726/esbuild.config.mjs#L60

miaobuao commented:
@joethei It appears that all the problems have been addressed. When you have a moment, could you kindly continue to move this issue forward? Thank you!

kevindenight commented:
@joethei I'm having some trouble with the remotely-save plugin, and I'm very eager to try this new plugin to see if it can help me solve my problem. Could you please expedite the review process?

shiqiFU0517 commented:
I'd really like to try this new plugin, could you please expedite the review?Thanks.

kris-teng commented:
真的很需要这个插件，求火速上架！

ZhangTaihong commented:
快快快！

huaingabc commented:
急需这个插件，建议开发人员别下班了！（bushi）

ZaoDee commented:
真的很需要这个插件，求火速上架！

zhushizhou commented:
好东西，搞快点！GKD！GKD！

xiaohe19 commented:
真的很需要这个插件，求火速上架！

demonq0q commented:
真的很需要这个插件，求火速上架！

sitdownkevin commented:
Really need this

mirrorxxx commented:
真的很需要这个插件，求火速上架！

Wangkejun520 commented:
真的很需要这个插件，求火速上架！

lxk1008 commented:
急需急需，太需要这个插件了，盼望抓紧上架。

yenan888 commented:
求ob的多端同步呀，很需要

1102979802 commented:
真的很需要这个插件，求火速上架！

engvuchen commented:
真的很需要这个插件，求火速上架！

anyish commented:
真的很需要这个插件，求火速上架！

Lion176 commented:
光速上架！快快快！

youcanfind commented:
真的很需要这个插件，快点上架啊！

whhwch1986 commented:
急需急需，太需要这个插件了，盼望抓紧上架。

YYW10086334 commented:
能不能快点上架啊，急需

HaibaraChyan commented:
同求，太需要了

Xuexu-Wang commented:
急需该插件，请求赶快上架呀～

joethei commented:
I am sorry for locking this PR, but we don't tolerate this kind of behavior.
I am going to close this PR now, feel free to resubmit, but please refrain from asking others to spam the PR, or you will be blocked from submitting plugins altogether.
